src/libcore/mem/maybe_uninit.rs

```diff
@@ -402,6 +403,14 @@ impl<T> MaybeUninit<T> {
     ///
     /// [inv]: #initialization-invariant
     ///
+    /// On top of that, remember that most types have additional invariants beyond merely
+    /// being considered initialized at the type level. For example, a `1`-initialized [`Vec<T>`]
+    /// is considered initialized (under the current implementation, this does not constitute
```

Comment:
Grammatical note: there are two possible interpretations with the comma here. If you change the comma to a semicolon `;` or double dash `--`, is it the meaning you intended? If so, it will be better to change it. If not, it will be better to rephrase.

Another option is to drop the "this does not constitute a stable guarantee" part of the sentence because it is redundant with "under the current implementation".

Reply:
> Another option is to drop the "this does not constitute a stable guarantee" part of the sentence because it is redundant with "under the current implementation".

It's not; it is making the point that the current implementation is not a stable guarantee, but I'd rephrase as: "under the current implementation, which does not constitute a stable guarantee".

Reply:
@bluetech thanks! Turned it into a `;`